src/floyd-warshall.jl

```diff
@@ -36,17 +36,17 @@ end
 #     Note that it is possible to consume large amounts of memory as the
 #     space required for the FloydWarshallState is O(n^2).
 #     """ ->
-function floyd_warshall_shortest_paths(
+function floyd_warshall_shortest_paths{T<:Real}(
     g::AbstractGraph;
-    edge_dists::AbstractArray{Float64, 2} = Array(Float64,(0,0))
+    edge_dists::AbstractArray{T,2} = zeros(0,0)
 )
 
     use_dists = issparse(edge_dists)? nnz(edge_dists > 0) : !isempty(edge_dists)
     n_v = nv(g)
-    dists = fill(convert(Float64,Inf), (n_v,n_v))
-    parents = zeros(Int, (n_v,n_v))
+    S = typeof(one(T) + one(T))
+    dists = fill(typemax(S), n_v, n_v)
```

Comment:
I do like the approach taken in LightGraphs.jl, since I usually don't have much metadata to carry around with the graphs I'm interested in. For this reason, I strongly urge you to rethink your opposition to type parameters. You are throwing away the single largest benefit of using Julia for what I consider to be a non-issue. The benefit here is to express the polymorphism that is natural based on user input. Again, the use of type parameters here should be distinguished from the Graphs.jl approach, which forces uses to think about types for associated metadata even if the user doesn't want to. Here, the type parameter is simply capturing some information that users have already provided. (Even if they don't specify it, the element type is implicit in the array.) In contrast, forcing users to use only `Float64`s for distances is a very un-Julian idiom that is only used in `Base` for floating point code that needs to know about the bit length.

This change uses `typemax(T)` as the sentinel value that represents "no edge exists". Note that this generalizes the use of `+Inf` for floats since `typemax(Float64) == +Inf`. I've also guarded against overflows like `typemax(T) + 3` explicitly, and put in the accompanying test changes that it works on a variety of element types for the distances. Now granted that overflow arithmetic will mean that users may get wrong answers for `UInt8` weights on graphs with > 256 nodes, but consider that the casual user will _not_ construct such weights (you have to explicitly type the weight matrix to contain `UInt8`s) and the arithmetic has the benefit of being exact in more cases.